c7n/resources/aws.py

```diff
@@ -541,9 +541,15 @@ def initialize_policies(self, policy_collection, options):
         service_region_map, resource_service_map = get_service_region_map(
             options.regions, policy_collection.resource_types)
         if 'all' in options.regions:
+            session = get_profile_session(options)
+            if session.region_name is None:
+                raise InvalidCliOption(
+                    'using `--region=all` needs a default region '
+                    'environment variable (AWS_DEFAULT_REGION) '
+                    'or a aws cli profile region set, to '
```

Comment:
`an aws cli`.. although perhaps just use the wording from the standard cli error message

`No default region set. Specify a default via AWS_DEFAULT_REGION or setting a region in ~/.aws/config`